pennylane/labs/resource_estimation/ops/op_math/symbolic.py

```diff
@@ -0,0 +1,217 @@
+            "num_work_wires": len(self.work_wires),
+        }
+
+    @classmethod
+    def resource_rep(
+        cls, base_class, base_params, num_ctrl_wires, num_ctrl_values, num_work_wires
+    ) -> re.CompressedResourceOp:
+        return re.CompressedResourceOp(
+            cls,
+            {
+                "base_class": base_class,
+                "base_params": base_params,
+                "num_ctrl_wires": num_ctrl_wires,
+                "num_ctrl_values": num_ctrl_values,
+                "num_work_wires": num_work_wires,
+            },
+        )
+
+    @classmethod
+    def controlled_resource_decomp(
+        cls,
+        outer_num_ctrl_wires,
+        outer_num_ctrl_values,
+        outer_num_work_wires,
+        base_class,
+        base_params,
+        num_ctrl_wires,
+        num_ctrl_values,
+        num_work_wires,
+    ) -> Dict[re.CompressedResourceOp, int]:
+        return {
+            cls.resource_rep(
+                base_class,
+                base_params,
+                outer_num_ctrl_wires + num_ctrl_wires,
+                outer_num_ctrl_values + num_ctrl_values,
+                outer_num_work_wires + num_work_wires,
+            ): 1
+        }
+
+    @staticmethod
+    def tracking_name(base_class, base_params, num_ctrl_wires, num_ctrl_values, num_work_wires):
+        base_name = base_class.tracking_name(**base_params)
+        return f"C({base_name},{num_ctrl_wires},{num_ctrl_values},{num_work_wires})"
+
+
+class ResourcePow(PowOperation, re.ResourceOperator):
+    """Resource class for Pow"""
+
+    @staticmethod
+    def _resource_decomp(
+        base_class, z, base_params, **kwargs
+    ) -> Dict[re.CompressedResourceOp, int]:
+        try:
+            return base_class.pow_resource_decomp(z, **base_params)
+        except re.ResourcesNotDefined:
+            pass
+
+        try:
+            return _scale_dict(base_class.resources(**base_params, **kwargs), z)
+        except re.ResourcesNotDefined:
+            pass
+
+        return {base_class.resource_rep(): z}
+
+    def resource_params(self) -> dict:
+        return {
+            "base_class": type(self.base),
+            "z": self.z,
+            "base_params": self.base.resource_params(),
+        }
+
+    @classmethod
+    def resource_rep(cls, base_class, z, base_params, **kwargs) -> re.CompressedResourceOp:
+        return re.CompressedResourceOp(
+            cls, {"base_class": base_class, "z": z, "base_params": base_params}
+        )
+
+    @classmethod
+    def pow_resource_decomp(
+        cls, z0, base_class, z, base_params, **kwargs
+    ) -> Dict[re.CompressedResourceOp, int]:
+        return {cls.resource_rep(base_class, z0 * z, base_params): 1}
+
+    @staticmethod
+    def tracking_name(base_class, z, base_params) -> str:
+        base_name = base_class.tracking_name(**base_params)
+        return f"({base_name})**{z}"
+
+
+def zyz_resources(num_ctrl_wires, num_work_wires):
```

Comment:
There are no tests for this. Also how is this going to be used?

Reply:
I'm not sure yet. I wrote this since zyz is used in the `qml.ctrl` logic, but I don't think it applies to any of the operators we have implemented so far. Maybe it should just be removed from this PR.